apps/web/src/studio/components/workflows/preferences/WorkflowGeneralSettingsForm.tsx

```diff
@@ -0,0 +1,48 @@
+import { useClipboard } from '@mantine/hooks';
+import { IconButton, Input, Textarea } from '@novu/novui';
+import { IconCheck, IconCopyAll } from '@novu/novui/icons';
+import { Stack } from '@novu/novui/jsx';
+import { ChangeEvent, FC } from 'react';
+import { WorkflowGeneralSettings } from './types';
+
+export type WorkflowGeneralSettingsProps = {
+  settings: WorkflowGeneralSettings;
+  updateSettings: (settings: WorkflowGeneralSettings) => void;
+  areSettingsDisabled?: boolean;
+};
+
+export const WorkflowGeneralSettingsForm: FC<WorkflowGeneralSettingsProps> = ({
+  settings,
+  updateSettings,
+  areSettingsDisabled,
+}) => {
+  const { copied, copy } = useClipboard();
+  const onChange =
+    (fieldKey: keyof WorkflowGeneralSettings) => (event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+      const value = event.target.value;
+      updateSettings({ ...settings, [fieldKey]: value });
+    };
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** why aren't we using a controller on each of these fields separately? What is the reason behind this single onChange?‏

Reply:
Because I originally tried to write this component to be independent of `react-hook-form` and its management, and then be wrapped in a Controller that would inject this behavior. I will update this tomorrow when I'm online